lib/vnet/user_process_windows_test.go

```diff
@@ -0,0 +1,49 @@
+// Teleport
+// Copyright (C) 2025 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package vnet
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+// TestSecureCredDir tests that the files can be created in a directory
+// secured by secureCredDir, and they are not readable by the current user.
+func TestSecureCredDir(t *testing.T) {
+	creds, err := newIPCCredentials()
+	require.NoError(t, err)
+
+	// Can't use t.TempDir() because we actually can't delete these files
+	// and it will fail the test. Note: these files leak.
+	credPath, err := os.MkdirTemp("", "test_creds")
+	require.NoError(t, err)
+	require.NoError(t, secureCredDir(credPath))
+	require.NoError(t, creds.client.write(credPath))
+
+	_, err = readCredentials(credPath)
+	require.Error(t, err, "expected no access to creds")
+
+	files := []string{certFileName, keyFileName, caFileName}
+	for _, f := range files {
+		fp := filepath.Join(credPath, f)
+		_, err := os.ReadFile(fp)
+		require.Error(t, err, "expected no access to %s", fp)
```

Comment:
Same for checking for a specific error. Also, Keep Going.

* https://go.dev/wiki/TestComments#keep-going